.github/workflows/docs/build-docs

```diff
@@ -80,56 +80,14 @@ def build_module_docs():
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(
-        description="Build HTML documentation for one or more modules."
+        description="Build HTML documentation for pytket-quantinuum."
     )
-    parser.add_argument(
-        "-m",
-        "--modules",
-        nargs="*",
-        default=[],
-        help="names of modules to build (without the `pytket-` prefix)",
-    )
-    parser.add_argument("-a", "--all", action="store_true", help="build all modules")
-    parser.add_argument("-i", "--index", action="store_true", help="build index page")
     parser.add_argument("-d", "--dest", help="copy artifacts into destination folder")
     args = parser.parse_args()
 
     print("Building docs for modules:", MODULE)
     build_module_docs()
 
-    if args.index:
-        print("Building index page...")
-        index_rst = DOCS_DIR / "index.rst"
-        with open(DOCS_DIR / "intro.txt", "r") as f:
-            content = f.readlines()
-            content.append(
-                "\n.. toctree::\n\t:caption: More documentation:\n\t:maxdepth: 1\n\n"
-            )            
-            content.append(f"\tpytket <{PYTKET_DOCS_LINK}>\n")
-            content.append(f"\tpytket extensions <{PYTKET_EX_DOCS_LINK}>\n")
```

Comment:
There is still some clean-up that could be done, e.g. `PYTKET_EX_DOCS_LINK` is no longer needed, though this could be deferred to another PR.

Reply:
I think this link is still needed? It is still used in line 51, adding the link to the overview of all extensions, see: https://cqcl.github.io/pytket-quantinuum/api/